src/build/app/mod.rs

```diff
@@ -2225,7 +2225,13 @@ impl<'help> App<'help> {
 
         // do the real parsing
         let mut parser = Parser::new(self);
-        parser.get_matches_with(&mut matcher, it)?;
+        if let Err(error) = parser.get_matches_with(&mut matcher, it) {
+            if self.is_set(AppSettings::IgnoreErrors) {
+                debug!("App::_do_parse: ignoring error: {}", error);
+            } else {
+                return Err(error);
+            }
+        }
```

Comment:
Have you looked into all the edge cases where the next few lines can fail if we don't return err here? Let's safeguard that.

Reply:
I admittedly couldn't review all edge cases for the next few lines comprehensively. They don't return errors. Are you concerned about panicking or unexpected behavior?

Reply:
And what do you mean exactly with "Let's safeguard this?" (just to avoid comment round-trips)

Thank you!

Reply:
I think it's okay.